sqlglotrs/src/token.rs

```diff
@@ -0,0 +1,428 @@
+    CONNECT_BY,
+    CONSTRAINT,
+    CREATE,
+    CROSS,
+    CUBE,
+    CURRENT_DATE,
+    CURRENT_DATETIME,
+    CURRENT_TIME,
+    CURRENT_TIMESTAMP,
+    CURRENT_USER,
+    DEFAULT,
+    DELETE,
+    DESC,
+    DESCRIBE,
+    DICTIONARY,
+    DISTINCT,
+    DISTRIBUTE_BY,
+    DIV,
+    DROP,
+    ELSE,
+    END,
+    ESCAPE,
+    EXCEPT,
+    EXECUTE,
+    EXISTS,
+    FALSE,
+    FETCH,
+    FILTER,
+    FINAL,
+    FIRST,
+    FOR,
+    FORCE,
+    FOREIGN_KEY,
+    FORMAT,
+    FROM,
+    FULL,
+    FUNCTION,
+    GLOB,
+    GLOBAL,
+    GROUP_BY,
+    GROUPING_SETS,
+    HAVING,
+    HINT,
+    IGNORE,
+    ILIKE,
+    ILIKE_ANY,
+    IN,
+    INDEX,
+    INNER,
+    INSERT,
+    INTERSECT,
+    INTERVAL,
+    INTO,
+    INTRODUCER,
+    IRLIKE,
+    IS,
+    ISNULL,
+    JOIN,
+    JOIN_MARKER,
+    KEEP,
+    KILL,
+    LANGUAGE,
+    LATERAL,
+    LEFT,
+    LIKE,
+    LIKE_ANY,
+    LIMIT,
+    LOAD,
+    LOCK,
+    MAP,
+    MATCH_RECOGNIZE,
+    MEMBER_OF,
+    MERGE,
+    MOD,
+    MODEL,
+    NATURAL,
+    NEXT,
+    NOTNULL,
+    NULL,
+    OBJECT_IDENTIFIER,
+    OFFSET,
+    ON,
+    ORDER_BY,
+    ORDERED,
+    ORDINALITY,
+    OUTER,
+    OVER,
+    OVERLAPS,
+    OVERWRITE,
+    PARTITION,
+    PARTITION_BY,
+    PERCENT,
+    PIVOT,
+    PLACEHOLDER,
+    PRAGMA,
+    PRIMARY_KEY,
+    PROCEDURE,
+    PROPERTIES,
+    PSEUDO_TYPE,
+    QUALIFY,
+    QUOTE,
+    RANGE,
+    RECURSIVE,
+    REPLACE,
+    RETURNING,
+    REFERENCES,
+    RIGHT,
+    RLIKE,
+    ROLLBACK,
+    ROLLUP,
+    ROW,
+    ROWS,
+    SELECT,
+    SEMI,
+    SEPARATOR,
+    SERDE_PROPERTIES,
+    SET,
+    SETTINGS,
+    SHOW,
+    SIMILAR_TO,
+    SOME,
+    SORT_BY,
+    START_WITH,
+    STRUCT,
+    TABLE_SAMPLE,
+    TEMPORARY,
+    TOP,
+    THEN,
+    TRUE,
+    UNCACHE,
+    UNION,
+    UNNEST,
+    UNPIVOT,
+    UPDATE,
+    USE,
+    USING,
+    VALUES,
+    VIEW,
+    VOLATILE,
+    WHEN,
+    WHERE,
+    WINDOW,
+    WITH,
+    UNIQUE,
+    VERSION_SNAPSHOT,
+    TIMESTAMP_SNAPSHOT,
+}
+
+#[pymethods]
+impl TokenType {
+    #[pyo3(name = "__repr__")]
+    fn python_repr(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self))
+    }
+}
+
+#[derive(Debug)]
+#[pyclass]
+pub struct Token {
+    #[pyo3(get)]
+    pub token_type: TokenType,
+    #[pyo3(get)]
+    pub text: String,
+    #[pyo3(get)]
+    pub line: usize,
+    #[pyo3(get)]
+    pub column: usize,
+    #[pyo3(get)]
+    pub start: usize,
+    #[pyo3(get)]
+    pub end: usize,
+    #[pyo3(get)]
+    pub comments: Vec<String>,
+}
+
+impl Token {
+    pub fn new(
+        token_type: TokenType,
+        text: String,
+        line: usize,
+        column: usize,
+        start: usize,
+        end: usize,
+        comments: Vec<String>,
+    ) -> Token {
+        Token {
+            token_type,
+            text,
+            line,
+            column,
+            start,
+            end,
+            comments,
+        }
+    }
+
+    pub fn number(num: u32) -> Token {
```

Comment:
I don't think these are used in the python Token class? Might be worth leaving out here.

Reply:
Yeah, good point.